opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -164,24 +165,32 @@ def upload_stats(self, view_data):
         """ It receives an array of view_data object
             and create time series for each value
         """
-        time_series_lists = self.make_request(view_data, MAX_TIME_SERIES_PER_UPLOAD)
-        for time_series_list in time_series_lists:
+        time_series_batches = self.create_batched_time_series(
+            view_data, MAX_TIME_SERIES_PER_UPLOAD)
+        for time_series_batch in time_series_batches:
             self.client.create_time_series(
                 self.client.project_path(self.options.project_id),
-                time_series_list)
+                time_series_batch)
 
-    def make_request(self, view_data, limit):
+    def create_batched_time_series(self, view_data, batch_size):
```

Comment:
I think changing this to use an iterator is a good idea in general, but I'd save the change for a different PR and only include the changes you need to fix the exporter here. It would make the review quicker and reduce the risk of regressions.